views/main.erb

```diff
@@ -59,7 +59,7 @@ function handleSubmit()
 
 document.getElementById('get').onclick = function() {
     xmlHttp = new XMLHttpRequest();
-    xmlHttp.open( "GET", "http://"+window.location.hostname+":"+window.location.port+"/keys/"+document.getElementById("key").value, false );
+    xmlHttp.open( "GET", "https://"+window.location.hostname+":"+window.location.port+"/keys/"+document.getElementById("key").value, false );
```

Comment:
this will break when served over http. use `window.location.protocol+"//"+window.location.hostname...`